src/app/aliases/aliases.lister.ts

```diff
@@ -0,0 +1,162 @@
+  MatSelectModule,
+  MatSnackBarModule,
+  MatTableModule,
+  MatTabsModule,
+  MatChipsModule,
+  MatDialog,
+  MatPaginator,
+  MatSnackBar,
+} from '@angular/material';
+import {MatFormFieldModule} from '@angular/material/form-field'; 
+import {AliasesEditorModal} from '../aliases/aliases.editor.modal';
+import {RMM} from '../rmm';
+
+@Component({
+    selector: 'aliases-lister',
+    styles: [`
+        .aliases-lister form > div {
+            display: inline-block;
+            position: relative;
+            margin: 10px;
+        }
+        mat-divider.transparent {
+            border-color: transparent;
+        }
+    `],
+    template: `
+    <div class="aliases-lister">
+        <ng-content select="[section-header]" style="margin-top: 20px;"></ng-content>
+        <ng-content select="[section-description]"></ng-content>
+        <ng-content select="[section-buttons]"></ng-content>
+        <form *ngFor="let item of values; let i = index;">
+            <div style='width: 100%'>
+                <mat-divider *ngIf="i > 0"></mat-divider>
+
+                <mat-form-field class="alias" style="margin: 10px;">
+                    <input
+                        matInput
+                        placeholder="Alias"
+                        readonly="true"
+                        [value]="item.localpart + '@' + item.domain"
+                    >
+                </mat-form-field>
+
+                <mat-form-field class="forward_to" style="margin: 10px;">
+                    <input
+                        matInput
+                        placeholder="Deliver to Account"
+                        readonly="true"
+                        [value]="item.forward_to"
+                    >
+                </mat-form-field>
+<!--
+                <button 
+                    (click)="edit(item)" 
+                    color='primary' 
+                    mat-raised-button 
+                    style="margin: 10px;"
+                >
+                    Edit
+                </button>
+
+                <button 
+                    (click)="delete(i, item)"
+                    *ngIf="!is_delete_disabled"
+                    color='primary' 
+                    mat-raised-button
+                    style="margin: 10px;"
+                >
+                    Delete
+                </button>
+-->
+            </div>
+            <mat-divider class='transparent'></mat-divider>
+        </form>
+    </div>
+
+        `
+})
+export class AliasesLister {
+  @Input() values: any[];
+  @Input() is_delete_disabled: false;
+  @Output() ev_reload = new EventEmitter<string>();
+  private dialog_ref : any;
+  visible_code_check = {};
+  constructor(public dialog: MatDialog,
+    public snackBar: MatSnackBar,
+    public rmm: RMM,
+  ) {}
+  edit (item): void {
+      item = JSON.parse(JSON.stringify(item))
+      this.dialog_ref = this.dialog.open(AliasesEditorModal, {
+          width: '600px',
+          data: item
```

Comment:
Follow up on the above: here it could be ```{ width: '600px', data: { item: item, mode: EditorMode.Update }```. It is then impossible to forget/fail to set the right mode (as long as the constructor remembers to complain about missing mode :))

Reply:
resolved